File: screens/RecommendationScreen.js
import React from 'react';
import {
  View,
  Text,
  TouchableOpacity,
  StyleSheet,
  ScrollView,
} from 'react-native';

export default function RecommendationScreen({ navigation }) {
  const recommendations = [
    {
      id: 1,
      name: 'Grilled Salmon',
      category: 'Seafood',
      rating: 4.7,
      reason: 'Based on your love for healthy options',
    },
    {
      id: 2,
      name: 'Chicken Tikka Masala',
      category: 'Indian',
      rating: 4.6,
      reason: 'Popular in your area',
    },
    {
      id: 3,
      name: 'Greek Salad',
      category: 'Mediterranean',
      rating: 4.5,
      reason: 'Perfect for lunch',
    },
    {
      id: 4,
      name: 'Chocolate Lava Cake',
      category: 'Dessert',
      rating: 4.9,
      reason: 'Highly rated dessert',
    },
    {
      id: 5,
      name: 'Ramen Bowl',
      category: 'Japanese',
      rating: 4.8,
      reason: 'Trending this week',
    },
  ];

  return (
    <View style={styles.container}>
      <Text style={styles.title}>Recommended for You</Text>
      <Text style={styles.subtitle}>Foods you might enjoy</Text>

      <ScrollView style={styles.list}>
        {recommendations.map((food) => (
          <TouchableOpacity
            key={food.id}
            style={styles.foodItem}
            onPress={() => navigation.navigate('Details', { food })}
          >
            <View style={styles.foodInfo}>
              <Text style={styles.foodName}>{food.name}</Text>
              <Text style={styles.foodCategory}>{food.category}</Text>
              <Text style={styles.foodReason}>💡 {food.reason}</Text>
            </View>
            <View style={styles.ratingContainer}>
              <Text style={styles.foodRating}>⭐ {food.rating}</Text>
              <Text style={styles.arrow}>→</Text>
            </View>
          </TouchableOpacity>
        ))}
      </ScrollView>

      <TouchableOpacity
        style={styles.backButton}
        onPress={() => navigation.navigate('Home')}
      >
        <Text style={styles.backButtonText}>← Back to Home</Text>
      </TouchableOpacity>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    padding: 20,
  },
  title: {
    fontSize: 28,
    fontWeight: 'bold',
    marginBottom: 5,
    color: '#333',
  },
  subtitle: {
    fontSize: 16,
    marginBottom: 20,
    color: '#666',
  },
  list: {
    flex: 1,
  },
  foodItem: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    padding: 15,
    backgroundColor: '#f9f9f9',
    borderRadius: 8,
    marginBottom: 10,
    borderLeftWidth: 4,
    borderLeftColor: '#4CAF50',
  },
  foodInfo: {
    flex: 1,
  },
  foodName: {
    fontSize: 18,
    fontWeight: 'bold',
    color: '#333',
  },
  foodCategory: {
    fontSize: 14,
    color: '#666',
    marginTop: 4,
  },
  foodReason: {
    fontSize: 12,
    color: '#4CAF50',
    marginTop: 6,
    fontStyle: 'italic',
  },
  ratingContainer: {
    alignItems: 'flex-end',
  },
  foodRating: {
    fontSize: 16,
    fontWeight: 'bold',
    color: '#FF6347',
  },
  arrow: {
    fontSize: 20,
    color: '#999',
    marginTop: 5,
  },
  backButton: {
    backgroundColor: '#FF6347',
    height: 50,
    borderRadius: 8,
    justifyContent: 'center',
    alignItems: 'center',
    marginTop: 20,
  },
  backButtonText: {
    color: '#fff',
    fontSize: 18,
    fontWeight: 'bold',
  },
});

File: flutter_app/lib/screens/recommendation_screen.dart
import 'package:flutter/material.dart';

class RecommendationScreen extends StatelessWidget {
  const RecommendationScreen({super.key});

  @override
  Widget build(BuildContext context) {
    final List<Map<String, dynamic>> recommendedFoods = [
      {
        'id': 1,
        'name': 'Chicken Tikka Masala',
        'category': 'Indian',
        'rating': 4.7,
        'reason': 'Based on your love for spicy foods'
      },
      {
        'id': 2,
        'name': 'Ramen Bowl',
        'category': 'Japanese',
        'rating': 4.6,
        'reason': 'You enjoyed similar noodle dishes'
      },
      {
        'id': 3,
        'name': 'Greek Salad',
        'category': 'Mediterranean',
        'rating': 4.4,
        'reason': 'Healthy option you might like'
      },
      {
        'id': 4,
        'name': 'Fish and Chips',
        'category': 'British',
        'rating': 4.3,
        'reason': 'Popular choice among food lovers'
      },
    ];

    return Scaffold(
      appBar: AppBar(
        title: const Text('Recommendations'),
        backgroundColor: Theme.of(context).colorScheme.inversePrimary,
      ),
      body: ListView.builder(
        padding: const EdgeInsets.all(20),
        itemCount: recommendedFoods.length,
        itemBuilder: (context, index) {
          final food = recommendedFoods[index];
          return Card(
            margin: const EdgeInsets.only(bottom: 15),
            child: Padding(
              padding: const EdgeInsets.all(12),
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: [
                  Row(
                    mainAxisAlignment: MainAxisAlignment.spaceBetween,
                    children: [
                      Expanded(
                        child: Text(
                          food['name'],
                          style: const TextStyle(
                            fontSize: 18,
                            fontWeight: FontWeight.bold,
                          ),
                        ),
                      ),
                      Row(
                        children: [
                          const Icon(Icons.star, color: Colors.amber, size: 20),
                          Text(
                            ' ${food['rating']}',
                            style: const TextStyle(
                              fontSize: 16,
                              fontWeight: FontWeight.bold,
                            ),
                          ),
                        ],
                      ),
                    ],
                  ),
                  const SizedBox(height: 5),
                  Text(
                    food['category'],
                    style: const TextStyle(
                      fontSize: 14,
                      color: Colors.grey,
                    ),
                  ),
                  const SizedBox(height: 10),
                  Row(
                    children: [
                      const Icon(
                        Icons.lightbulb_outline,
                        size: 16,
                        color: Colors.orange,
                      ),
                      const SizedBox(width: 5),
                      Expanded(
                        child: Text(
                          food['reason'],
                          style: const TextStyle(
                            fontSize: 14,
                            fontStyle: FontStyle.italic,
                            color: Colors.orange,
                          ),
                        ),
                      ),
                    ],
                  ),
                  const SizedBox(height: 10),
                  ElevatedButton(
                    onPressed: () {
                      Navigator.pushNamed(
                        context,
                        '/details',
                        arguments: food,
                      );
                    },
                    style: ElevatedButton.styleFrom(
                      backgroundColor: Colors.deepOrange,
                      foregroundColor: Colors.white,
                    ),
                    child: const Text('View Details'),
                  ),
                ],
              ),
            ),
          );
        },
      ),
    );
  }
}
